src/sentry/models/actor.py

```diff
@@ -113,8 +113,18 @@ def get_actor_id_for_user(user: Union["User", RpcUser]):
     if user.actor_id:
         return user.actor_id
     # Temporary Dual write
-    actor, _ = Actor.objects.get_or_create(type=ACTOR_TYPES["user"], user_id=user.id)
-    # TODO(hybrid-cloud): remove this after actor migration is complete
+    # Until we have indexes back online, we have to account for potential race condition on user creation
+    with transaction.atomic():
+        actors_for_user = Actor.objects.filter(type=ACTOR_TYPES["user"], user_id=user.id).all()
+        if actors_for_user == 1:
+            actor = actors_for_user[0]
+        else:
+            actor = Actor.objects.create(type=ACTOR_TYPES["user"], user_id=user.id)
```

Comment:
This will still raise an exception for users that have multiple currently.

Reply:
oh right, sorry, modifying the if condition and the test, good catch